### PR TITLE
fix: improvements for canceling workflow reminders

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Sendgrid doesn't allow more than 100 different batch ids in a 'paused' or 'pending cancellation' state at one time. I am assuming we sometimes reach that limit and that's why in some cases canceled bookings still send reminders. Currently, we have 88 batch ids in the queue, so pretty close to the limit already. 

With this PR we now schedule emails only 2 hours before they are scheduled instead of 72 hour before. This reduces the amount of scheduled emails that need to be canceled.

### CODE REVIEW

File: packages/features/ee/workflows/lib/reminders/emailReminderManager.ts [hunk 1]
```diff
@@ -288,7 +288,7 @@ export const scheduleEmailReminder = async (args: scheduleEmailReminderArgs) =>
     // Can only schedule at least 60 minutes and at most 72 hours in advance
     if (
       currentDate.isBefore(scheduledDate.subtract(1, "hour")) &&
-      !scheduledDate.isAfter(currentDate.add(72, "hour"))
+      !scheduledDate.isAfter(currentDate.add(2, "hour"))
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/lib/reminders/emailReminderManager.ts:291]: 2 hours should be enough, we are running the corn job to schedule unscheduled emails every 15 minutes

**Review by zomars — APPROVED** [review]

**Review by PeerRich — DISMISSED** [review]

**Comment by PeerRich** [comment]
clever fix

**Review by zomars — CHANGES_REQUESTED** [review]
@CarinaWolli seems like we need to update tests:

image

File: packages/features/ee/workflows/lib/reminders/emailReminderManager.ts [hunk 2]
```diff
@@ -327,8 +328,8 @@ export const scheduleEmailReminder = async (args: scheduleEmailReminderArgs) =>
       } catch (error) {
         log.error(`Error scheduling email with error ${error}`);
       }
-    } else if (scheduledDate.isAfter(currentDate.add(72, "hour"))) {
-      // Write to DB and send to CRON if scheduled reminder date is past 72 hours
+    } else if (scheduledDate.isAfter(currentDate.add(2, "hour"))) {
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/lib/reminders/emailReminderManager.ts:331]: forgot to change that, failed test caught that

File: packages/features/ee/workflows/lib/test/workflows.test.ts [hunk 3]
```diff
@@ -86,17 +86,17 @@ const mockBookings = [
     eventTypeId: 1,
     userId: 101,
     status: BookingStatus.ACCEPTED,
-    startTime: `2024-05-22T04:00:00.000Z`,
-    endTime: `2024-05-22T04:30:00.000Z`,
+    startTime: `2024-05-20T14:00:00.000Z`,
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/lib/test/workflows.test.ts:89]: make date closer to the current date `2024-05-20T11:59:59Z` so we can test if scheduling works

**Review by zomars — APPROVED** [review]

**Pull request merged**